x-pack/plugins/ingest_manager/server/services/agents/setup.ts

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { SavedObjectsClientContract } from 'src/core/server';
+import { SO_SEARCH_LIMIT } from '../../constants';
+import { agentPolicyService } from '../agent_policy';
+import { outputService } from '../output';
+import { getLatestConfigChangeAction } from './actions';
+
+export async function isAgentsSetup(soClient: SavedObjectsClientContract): Promise<boolean> {
+  const adminUser = await outputService.getAdminUser(soClient, false);
+  const outputId = await outputService.getDefaultOutputId(soClient);
+  // If admin user (fleet_enroll) and output id exist Agents are correctly setup
+  return adminUser !== null && outputId !== null;
```

Comment:
this condition seems to differ from the original, where we just check for truthiness. is there a specific need to change to check for null instead?

Reply:
no there is no specific need I check for null to fix a type issue here, but I could replace it by 
```
!!adminUser && !!outputId
```